### PR TITLE
feature - api shops

### PR DESCRIPTION
**Pull request by at-lytruong**

### CODE REVIEW

File: MyApp/Model/API/Api.Category.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
+////
+////  Api.Category.swift
+////  MyApp
+////
+////  Created by Hải Lý on 8/14/17.
+////  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+////
+//
+//import Foundation
```
Comment (at-daiho) [inline MyApp/Model/API/Api.Category.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.Category.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+//import Alamofire
```
Comment (at-daiho) [inline MyApp/Model/API/Api.Category.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.Category.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+//import ObjectMapper
```
Comment (at-daiho) [inline MyApp/Model/API/Api.Category.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.Category.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+//import RealmS
```
Comment (at-daiho) [inline MyApp/Model/API/Api.Category.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.Category.swift [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+//import RealmSwift
```
Comment (at-daiho) [inline MyApp/Model/API/Api.Category.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.Category.swift [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+//
+//extension Api.Category {
```
Comment (at-daiho) [inline MyApp/Model/API/Api.Category.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.MenuItem.swift [hunk 7]
```diff
@@ -0,0 +1,47 @@
+////
+////  Api.MenuItem.swift
+////  MyApp
+////
+////  Created by Hải Lý on 8/11/17.
+////  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+////
+//
+//import Foundation
```
Comment (at-daiho) [inline MyApp/Model/API/Api.MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.MenuItem.swift [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+//import Alamofire
```
Comment (at-daiho) [inline MyApp/Model/API/Api.MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.MenuItem.swift [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+//import ObjectMapper
```
Comment (at-daiho) [inline MyApp/Model/API/Api.MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.MenuItem.swift [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+//import RealmS
```
Comment (at-daiho) [inline MyApp/Model/API/Api.MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.MenuItem.swift [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+//import RealmSwift
```
Comment (at-daiho) [inline MyApp/Model/API/Api.MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.MenuItem.swift [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+//
+//extension Api.MenuItem {
```
Comment (at-daiho) [inline MyApp/Model/API/Api.MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/API/Api.Menus.swift [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+import Foundation
+import Alamofire
+import ObjectMapper
+import RealmS
+import RealmSwift
+
+extension Api.Menus {
+        struct QueryParams {
+            let id: Int
+
```
Comment (at-daiho) [inline MyApp/Model/API/Api.Menus.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/Model/Schema/MenuItem.swift [hunk 14]
```diff
@@ -0,0 +1,76 @@
+//
+//  MenuItem.swift
+//  MyApp
+//
+//  Created by Hải Lý on 8/11/17.
+//  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+//
+
```
Comment (at-daiho) [inline MyApp/Model/Schema/MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/Schema/MenuItem.swift [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+//import Foundation
+//import ObjectMapper
```
Comment (at-daiho) [inline MyApp/Model/Schema/MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/Schema/MenuItem.swift [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+//import RealmS
```
Comment (at-daiho) [inline MyApp/Model/Schema/MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/Schema/MenuItem.swift [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+//import RealmSwift
```
Comment (at-daiho) [inline MyApp/Model/Schema/MenuItem.swift]: WARNING - There should be a space after //

File: MyApp/Model/Schema/MenuItem.swift [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+//
+//final class MenuItem: Object, StaticMappable {
```
Comment (at-daiho) [inline MyApp/Model/Schema/MenuItem.swift]: WARNING - There should be a space after //

**Comment by at-daiho** [comment]
Total linter violations in pull request: 18
Serious: 0
Warnings: 18

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/Define/Demo.swift [hunk 19]
```diff
@@ -7,19 +7,33 @@
...
     }
+}
 
 struct Profile {
     var icon = ""
     var nameIcon = ""
 }
-struct Category {
+
+struct Category1 {
```
Comment (at-hieutran) [inline MyApp/Define/Demo.swift]: Tên gì đây. Đổi lại cái khác

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 20]
```diff
@@ -23,14 +24,31 @@ final class HomeViewController: ViewController {
...
+        }
+    }
+
+    func updateView() {
+        guard isViewLoaded else { return }
+        listCollectionView.reloadData()
     }
 
     func configViewModel() {
+        viewModel.delegate2 = self
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: delegate2 là gi đấy ???

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 21]
```diff
@@ -48,14 +66,15 @@ final class HomeViewController: ViewController {
...
+        scrollView.contentSize = CGSize(width: UIScreen.main.bounds.width * CGFloat(categories.count), height: UIScreen.main.bounds.height * 250 / 667)
 
-        for index in 0..<viewModel.items.count {
+        for index in 0..<categories.count {
             frame.size.width = UIScreen.main.bounds.width
             frame.size.height = UIScreen.main.bounds.height * 250 / 667
             frame.origin.x = UIScreen.main.bounds.width * CGFloat(index)
             let subView = UIImageView(frame: frame)
-            subView.image = UIImage(named: viewModel.items[index].image)
+//                        subView.image = UIImage(named: categories[index].image)
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: k dùng thì xoá đi

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 22]
```diff
@@ -74,8 +93,9 @@ final class HomeViewController: ViewController {
 
     @objc private func imageTapped(_ sender: AnyObject) {
         let vc = DetailViewController()
-        let item = viewModel.items[sender.view.tag]
-        vc.viewModel = viewModel.detailViewModelWith(item: item)
+//        guard let categories = viewModel.categories else { return }
+//        let item = categories[sender.view.tag]
+        //        vc.viewModel = viewModel.detailViewModelWith(item: item)
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: k dùng thì xoá

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 23]
```diff
@@ -131,7 +151,8 @@ extension HomeViewController: UICollectionViewDataSource {
 extension HomeViewController: UICollectionViewDelegate {
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
         let vc = ListItemViewController()
-        vc.category = viewModel.categories[indexPath.row].category
+//        guard let categories = viewModel.categories else { return }
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: xoá

File: MyApp/View/Controllers/ListItem/TableViewCell/ListItemTableViewCell.swift [hunk 24]
```diff
@@ -22,7 +23,8 @@ final class ListItemTableViewCell: UITableViewCell {
 
     func updateView() {
         guard let viewModel = viewModel else { return }
-        foodImageView.image = UIImage(named: viewModel.image)
+        let url = URL(string: viewModel.image)
+        foodImageView.sd_setImage(with: url)
```
Comment (at-hieutran) [inline MyApp/View/Controllers/ListItem/TableViewCell/ListItemTableViewCell.swift]: gắn thêm planholder khi load ảnh

File: MyApp/View/Controllers/favorite/FavoriteCell.swift [hunk 25]
```diff
@@ -23,7 +23,8 @@ class FavoriteCell: UITableViewCell {
 
     private func updateView() {
         guard let viewModel = viewModel else { return }
-        favoriteImageView.image = UIImage(named: viewModel.image)
+        let url = URL(string: viewModel.image)
+        favoriteImageView.sd_setImage(with: url)
```
Comment (at-hieutran) [inline MyApp/View/Controllers/favorite/FavoriteCell.swift]: gắn thêm planholder khi load ảnh

File: MyApp/ViewModel/Detail/DetailViewModel.swift [hunk 26]
```diff
@@ -19,6 +19,8 @@ final class DetailViewModel: MVVM.ViewModel {
     var price = 0
     var category = ""
 
+//    typealias Item = App.Demo.Item
```
Comment (at-hieutran) [inline MyApp/ViewModel/Detail/DetailViewModel.swift]: xoá

File: MyApp/ViewModel/Favorite/FavoriteCellViewModel.swift [hunk 27]
```diff
@@ -21,6 +21,9 @@ final class FavoriteCellViewModel: MVVM.ViewModel {
 
     init(favoriteItem: FavoriteItem) {
         id = favoriteItem.id
+//        var url = URL(string: image)
+//        let urlImage = URL(string: favoriteItem.image)
+//        url = urlImage
```
Comment (at-hieutran) [inline MyApp/ViewModel/Favorite/FavoriteCellViewModel.swift]: xoá

File: MyApp/ViewModel/Home/HomeViewModel.swift [hunk 28]
```diff
@@ -8,62 +8,91 @@
...
 
+    var categories: Results<Category>?
+    var token: NotificationToken?
+
     enum Action {
         case update(Int)
     }
 
-    weak var delegate: HomeViewModelDelegate?
+    weak var delegate2: HomeViewModelDelegate?
```
Comment (at-hieutran) [inline MyApp/ViewModel/Home/HomeViewModel.swift]: đặt lại tên

File: MyApp/ViewModel/ListItem/ListItemViewModel.swift [hunk 29]
```diff
@@ -8,28 +8,56 @@
...
         return DetailViewModel(item: item)
     }
+
+//    func fetch() {
+//        items = RealmS().objects(MenuItem.self).sorted(byKeyPath: "id", ascending: true)
+//        token = items?.addNotificationBlock({ [weak self] (change) in
+//            guard let this = self else { return }
+//            this.notify(change: change)
+//        })
+//    }
```
Comment (at-hieutran) [inline MyApp/ViewModel/ListItem/ListItemViewModel.swift]: xoá

**Comment by at-daiho** [comment]
Total linter violations in pull request: 1
Serious: 0
Warnings: 1

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

**Comment by at-hieutran** [comment]
LGTM

**Pull request merged**